test/framework/toolchain.py

```diff
@@ -339,6 +353,20 @@ def test_optarch_generic(self):
                         else:
                             self.assertFalse(generic_flags in tc.get_variable(var))
 
+    def test_optarch_aarch64_heuristic(self):
```

Comment:
@boegel Nope. Here is my current test (with a `GCCcore/6.2.0` module in place):
```
    def test_optarch_aarch64_heuristic(self):
        """Test whether AArch64 pre-GCC-6 optimal architecture flag heuristic works."""
        st.get_cpu_architecture = lambda: st.AARCH64
        st.get_cpu_model = lambda: 'ARM Cortex-A53'
        st.get_cpu_vendor = lambda: st.ARM
        tc = self.get_toolchain("GCC", version="4.7.2")
        tc.prepare()
        self.assertEqual(tc.COMPILER_OPTIMAL_ARCHITECTURE_OPTION[tc.arch], 'mcpu=cortex-a53')

        st.get_cpu_model = lambda: 'ARM Cortex-A53 + Cortex-A72'
        tc = self.get_toolchain("GCC", version="4.7.2")
        tc.prepare()
        self.assertEqual(tc.COMPILER_OPTIMAL_ARCHITECTURE_OPTION[tc.arch], 'mcpu=cortex-a72.cortex-a53')

        tc = self.get_toolchain("GCCcore", version="6.2.0")
        tc.prepare()
        self.assertEqual(tc.COMPILER_OPTIMAL_ARCHITECTURE_OPTION[tc.arch], 'mcpu=native')
```
I've added
```
    print(self.COMPILER_OPTIMAL_ARCHITECTURE_OPTION[systemtools.AARCH64])
```
to `_set_optimal_architecture` in `easybuild-framework/toolchains/compiler/gcc.py` right at the beginning after the check that we're on AArch64 (i.e., before modifying the default value).  And it prints
```
mcpu=native
mcpu=cortex-a53
mcpu=cortex-a72.cortex-a53
```
Calling `tc.set_options({})` right before `tc.prepare()` in the test doesn't change anything.  So it keeps the modified value even when creating a new instance. What am I doing wrong here?  I'm confused...